packages/core/src/crawlers/error_snapshotter.ts

```diff
@@ -0,0 +1,148 @@
+import crypto from 'node:crypto';
+import { resolve } from 'node:path';
+
+import { pathExistsSync } from 'fs-extra';
+
+import type { ErrnoException } from './error_tracker';
+import type { CrawlingContext } from '../crawlers/crawler_commons';
+import type { KeyValueStore } from '../storages';
+import type { PlaywrightCrawlingContext, PuppeteerCrawlingContext, PlaywrightPage, PuppeteerPage, SnapshotResult } from '../typedefs';
+
+/**
+ * ErrorSnapshotter class is used to capture a screenshot of the page and a snapshot of the HTML when an error occur during web crawling.
+ */
+export class ErrorSnapshotter {
+    static readonly MAX_ERROR_CHARACTERS = 30;
+    static readonly MAX_HASH_LENGTH = 30;
+    static readonly MAX_FILENAME_LENGTH = 250;
+    static readonly BASE_MESSAGE = 'An error occurred';
+    static readonly SNAPSHOT_PREFIX = 'ERROR_SNAPSHOT';
+
+    private KEY_VALUE_STORE_LOCAL_PATH: string;
+    private KEY_VALUE_PLATFORM_PATH :string;
+
+    constructor() {
+        // v3.0.0 used `crawlee_storage` as the default, we changed this in v3.0.1 to just `storage`,
+        // this function handles it without making BC breaks - it respects existing `crawlee_storage`
+        // directories, and uses the `storage` only if it's not there.
+        const defaultStorageDir = () => {
+            if (pathExistsSync(resolve('./crawlee_storage'))) {
+                return 'crawlee_storage';
+            }
+
+            return 'storage';
+        };
+
+        this.KEY_VALUE_PLATFORM_PATH = 'https://api.apify.com/v2/key-value-stores';
+        this.KEY_VALUE_STORE_LOCAL_PATH = `file://${process.env.PWD}/${defaultStorageDir()}/key_value_stores`;
```

Comment:
lets remove this, there is no point in supporting crawlee 3.0.0 when this feature will land in 3.10.0

```suggestion
        this.KEY_VALUE_PLATFORM_PATH = 'https://api.apify.com/v2/key-value-stores';
        this.KEY_VALUE_STORE_LOCAL_PATH = `file://${process.env.PWD}/storage/key_value_stores`;
```

i also very much dislike the fact that we hardcode some apify URL here (i dislike its here, its not relevant whether you get the value from env vars or hardcode it). why do you need to handle this manually? we already have abstractions in place, you should just use KVS and it will do something locally and something else on the platform, why is that not enough and you need to deal with different paths explicitly?

Reply:
I guess Hamza would do that if he would be more aware of how the storage classes work behind the scenes :) Can you just suggest what to use?

Reply:
i would remove this manual URL handling completely and just call KVS.setValue (and deal with the URLs elsewhere, e.g. via that `getPublicUrl` method in SDK (which we can add to crawlee too if we want it to provide URLs to the locally stored files)

Reply:
i can try to improve this myself if needed, would very much like to have this ready by the end of the day or tomorrow morning. also its probably not a blocker for merging, @HamzaAlwan please focus on the other parts if this one is not clear (but let's remove the `crawlee_storage` conditional handling as suggested).